tests/instancetype/instancetype.go

```diff
@@ -541,17 +541,18 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			vm := libvmi.NewVirtualMachine(vmi,
 				libvmi.WithInstancetype(instancetype.Name),
 				libvmi.WithPreference(preference.Name),
+				libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
 			)
 			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
+			By("Waiting for VM to be ready")
+			Eventually(matcher.ThisVM(vm), 360*time.Second, 1*time.Second).Should(matcher.BeReady())
+			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
-			By("Waiting for VirtualMachineInstancetypeSpec and VirtualMachinePreferenceSpec ControllerRevision to be referenced from the VirtualMachine")
-			Eventually(func(g Gomega) {
-				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-				g.Expect(err).ToNot(HaveOccurred())
-				g.Expect(vm.Spec.Instancetype.RevisionName).ToNot(BeEmpty())
-				g.Expect(vm.Spec.Preference.RevisionName).ToNot(BeEmpty())
-			}, timeout, 1*time.Second).Should(Succeed())
```

Comment:
FWIW the original code is likely going to result in fewer calls to the API compared to the `Eventually` call above waiting for the VM to be `Ready` as we populate `RevisionNames` earlier in the core sync logic of the VM controller.

We should really share this logic somewhere however as it's used everywhere in these tests.

Reply:
Actually i don't think this is true since[ we do this anyway](https://github.com/kubevirt/kubevirt/blob/ab298fbb064741ec543ac06cc92a502f7ddcdc06/tests/libvmops/lifecycle.go#L73) with the start function

Reply:
@lyarwood  When do we populate RevisionNames?

Reply:
@Barakmor1 early on in the VirtualMachine controller sync loop regardless of the VM actually running etc:

https://github.com/kubevirt/kubevirt/blob/f60dc67eda2799877d94cdbaf18fbfd850f6f8e2/pkg/virt-controller/watch/vm/vm.go#L3254-L3258

https://github.com/kubevirt/kubevirt/blob/f60dc67eda2799877d94cdbaf18fbfd850f6f8e2/pkg/instancetype/controller/vm/controller.go#L122-L127 (This code was recently refactored and moved out of the core controller but functionally remains the same).

Reply:
So, @lyarwood, if I understand correctly, we have a controller that modifies the VM spec, which represents the desired state after it's created. I’ve never heard of this before. Is it common for a controller to modify specs? I thought mutators and object creators were responsible for setting the spec. This is surprising to me.

FYI @xpivarc

Reply:
@Barakmor1 correct, it was a mistake with the original design and implementation that we are trying to [remove in what remains of the v1.5.0 cycle](https://github.com/kubevirt/community/pull/317). However even with these references moved into `Status` the controller will continue to create and reference `ControllerRevisions` as part of the initial sync loop so the original test asserting this behaviour against a stopped VM will still be valid.

Reply:
From my perspective, I noticed that spec validations were done within an `eventually` scope, which is unusual and not a good pattern. That's why I decided to remove the `eventually` scope here. If the instancetype and preference controllers don't follow best practices, I don't feel comfortable modifying these tests. I will update the PR to only modify the cases where we create a stopped VM and start it immediately. I hope these issues will be resolved soon. 
Thanks for the explanation.